rpcs3/util/cpu_stats.cpp

```diff
@@ -139,37 +151,35 @@ namespace utils
 			return;
 		}
 
-		PDH_FMT_COUNTERVALUE counterVal{};
-		DWORD dwBufferSize = 0; // Size of the pItems buffer
-		DWORD dwItemCount = 0;  // Number of items in the pItems buffer
-		PDH_FMT_COUNTERVALUE_ITEM *pItems = NULL; // Array of PDH_FMT_COUNTERVALUE_ITEM structures
+		DWORD dwBufferSize = 0; // Size of the items buffer
+		DWORD dwItemCount = 0;  // Number of items in the items buffer
 
-		status = PdhGetFormattedCounterArray(m_cpu_cores, PDH_FMT_DOUBLE, &dwBufferSize, &dwItemCount, pItems);
+		status = PdhGetFormattedCounterArray(m_cpu_cores, PDH_FMT_DOUBLE, &dwBufferSize, &dwItemCount, nullptr);
 		if (PDH_MORE_DATA == status)
 		{
-			pItems = (PDH_FMT_COUNTERVALUE_ITEM*)malloc(dwBufferSize);
-			if (pItems)
+			std::vector<PDH_FMT_COUNTERVALUE_ITEM> items(std::ceil(dwBufferSize / static_cast<double>(sizeof(PDH_FMT_COUNTERVALUE_ITEM))));
```

Comment:
dwBufferSize seems to have some padding. so dwItemCount cannot be used directly here.